packages/rpc-graphql/src/resolvers/resolve-info/account.ts

```diff
@@ -0,0 +1,95 @@
+import { Address } from '@solana/addresses';
+import { DataSlice } from '@solana/rpc-core';
+import { Commitment, Slot } from '@solana/rpc-types';
+import { ArgumentNode, GraphQLResolveInfo } from 'graphql';
+
+import { AccountLoaderArgs } from '../../loaders';
+import { getAllFieldNodesByName, isInlineFragmentPresent } from './resolve-info';
+
+function findArgumentNodeByName(argumentNodes: readonly ArgumentNode[], name: string): ArgumentNode | undefined {
+    return argumentNodes.find(argumentNode => argumentNode.name.value === name);
+}
+
+function parseAccountEncodingArgument(
+    argumentNodes: readonly ArgumentNode[],
+    variableValues: {
+        [variable: string]: unknown;
+    },
+): AccountLoaderArgs['encoding'] | undefined {
+    const argumentNode = findArgumentNodeByName(argumentNodes, 'encoding');
+    if (argumentNode) {
+        if (argumentNode.value.kind === 'EnumValue') {
+            if (argumentNode.value.value === 'BASE_58') {
+                return 'base58';
+            }
+            if (argumentNode.value.value === 'BASE_64') {
+                return 'base64';
+            }
+            if (argumentNode.value.value === 'BASE_64_ZSTD') {
+                return 'base64+zstd';
+            }
+        }
+        if (argumentNode.value.kind === 'Variable') {
+            return variableValues[argumentNode.value.name.value] as AccountLoaderArgs['encoding'];
+        }
+    } else {
+        return undefined;
+    }
+}
+
+function parseAccountDataSliceArgument(
+    argumentNodes: readonly ArgumentNode[],
+    variableValues: { [variable: string]: unknown },
+): AccountLoaderArgs['dataSlice'] {
+    const argumentNode = findArgumentNodeByName(argumentNodes, 'dataSlice');
+    if (argumentNode) {
+        if (argumentNode.value.kind === 'ObjectValue') {
+            const offsetArg = argumentNode.value.fields?.find(field => field.name.value === 'offset');
+            const lengthArg = argumentNode.value.fields?.find(field => field.name.value === 'length');
+            const length = lengthArg?.value.kind === 'IntValue' ? parseInt(lengthArg.value.value) : undefined;
+            const offset = offsetArg?.value.kind === 'IntValue' ? parseInt(offsetArg.value.value) : undefined;
+            return length !== undefined && length !== null && offset !== undefined && offset !== null
+                ? { length, offset }
+                : undefined;
+        }
+        if (argumentNode.value.kind === 'Variable') {
+            return variableValues[argumentNode.value.name.value] as DataSlice;
+        }
+    } else {
+        return undefined;
+    }
+}
+
+/**
+ * Build a set of account loader args by inspecting which fields have
+ * been requested in the query (ie. `data` or inline fragments).
+ */
+export function buildAccountLoaderArgSetFromResolveInfo(
```

Comment:
Right now, this misses all of the fields inside spread fragments, both ones that necessitate `jsonParsed` fetching _and_ ones that fetch data. You have to write your node visitor and have it _recurse_ into fragment spreads (non-inline fragments).

https://astexplorer.net/#/gist/a0ce58c7813b11e20472fc6d6bf5a6f2/ca707e66ebe2eec7fa4a5cf176d30f8881b0ab5d

Reply:
Done.